### PR TITLE
streamline the way we connect/ping to the server

### PR DESCRIPTION
**Pull request by evgeni**
as always, this is an example, I intend to adjust that across all (tested) modules

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like a good start.

File: modules/foreman_domain.py [hunk 1]
```diff
@@ -129,15 +129,10 @@ def main():
         supports_check_mode=True,
     )
 
-    (server_params, domain_dict, state) = module.parse_params()
+    (_, domain_dict, state) = module.parse_params()
```
Comment (mdellweg) [inline modules/foreman_domain.py]: Do we still want to expose the server_params to the individual modules?

Comment (evgeni) [reply modules/foreman_domain.py]: Nope. I've just left it here for now as the other modules are still using it

**Comment by evgeni** [comment]
updated all tested modules for the new API

**Pull request merged**